src/main/java/org/ensime/maven/plugins/ensime/EnsimeConfigGenerator.java

```diff
@@ -685,36 +685,39 @@ private void write(final String content, final File out) {
     } catch(IOException ioex) {}
   }
 
-  /**
-   * Generates configurations.
-   */
-  public void generate(final File out) {
-
-    String projectDir =
-      project.getBasedir().toPath().toAbsolutePath().toString();
+  public EnsimeConfig generateConfig() {
+    String projectDir = project.getBasedir().toPath().toAbsolutePath().toString();
 
     File cacheDir = new File(projectDir + SP + ".ensime_cache");
 
     List<EnsimeProject> subProjects = getEnsimeProjects();
 
     Map<String, EnsimeModule> modules =
-      subProjects.stream().collect(groupingBy(s -> s.getId().getProject()))
-      .entrySet().stream()
-      .collect(toMap(p -> p.getKey(), p -> ensimeProjectsToModule(p.getValue())));
+        subProjects.stream().collect(groupingBy(s -> s.getId().getProject()))
+            .entrySet().stream()
+            .collect(toMap(Map.Entry::getKey, p -> ensimeProjectsToModule(p.getValue())));
```

Comment:
Is it more idiomatic to use `Map.Entry::`??

Reply:
Yup, https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html suggests doing this if the lambda just calls a single method.

Reply:
Thanks, just wanted confirm.